### PR TITLE
fix(#2132): TableView accessibility review issues

### LINKED ISSUE #2132: TableView accessibility review issues
# 🐛 Bug Report 

1. "Select" and "Select All"  strings for labelling checkboxes should be localized.
2. In single selection mode, we set `visibility: hidden` on the Select All Checkbox, however, a `columnheader` must contain content to provide an accessibility name. Possible solutions are to either add a `VisuallyHidden` with "Select" as the content to the `columnheader`, or set the `role` for the `columnheader` to `gridcell`, when the Select All Checkbox is hidden.
3. Documentation examples include unlabelled icon Button controls for illustrative purposes. ideally we should demonstrate best practices for labelling these buttons.
4. Column header cell for `selectionMode="single"` should not have `aria-disabled="true"`.
5. With VoiceOver, NVDA and JAWS, nested column examples fail to announce nested columns correctly. For example, in [`/story/tableview--static-with-nested-columns`](https://reactspectrum.blob.core.windows.net/reactspectrum/924bb2f2b251bfe171cfd75440fbf9f0638b8836/storybook-17/iframe.html?id=tableview--static-with-nested-columns&providerSwitcher-locale=&providerSwitcher-theme=&providerSwitcher-scale=&providerSwitcher-toastPosition=bottom&viewMode=story), the first column after Select All, announces as "GROUP 1 TEST" and the next column announces as "GROUP 2 FOO" despite `aria-colspan` attributes indicating that the "TEST" column does not fall under a nested column heading, and that "FOO" and "BAR" should fall under the "GROUP 1" heading. This may be a screen reader/browser bug, reflecting a lack of support for `aria-colspan`:
   - https://bugs.webkit.org/show_bug.cgi?id=171165
   - https://bugs.chromium.org/p/chromium/issues/detail?id=543050
   - https://github.com/FreedomScientific/VFO-standards-support/issues?q=is%3Aissue+is%3Aopen+colspan
   - https://github.com/nvaccess/nvda/issues?q=is%3Aissue+is%3Aopen+colspan
6. Sort order change is not announced by screen readers when a sortable column header is clicked, and the current sort order state for the table cannot be determined when entering the table, without navigating through each column header. A suggested fix would be to use live region announcement to announce sort changes, and to use a `hidden` element referenced with `aria-describedby` to communicate the current sort order.

## 🔦 Context

These issues have been raised as part of an Accessibility review as we approach release for the TableView component.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | @react-spectrum/table@3.0.0-beta.1 
| Browser          | Chrome Version 91.0.4472.114 (Official Build) (x86_64)
| Operating System | MacOS 11.4 (20F71)

## 🧢 Your Company/Team

Adobe/Accessibility

### PR DESCRIPTION
**Pull request by majornista**
1. "Select" and "Select All" strings for labelling checkboxes should be localized.
2. In single selection mode, we set visibility: hidden on the Select All Checkbox, however, a columnheader must contain content to provide an accessibility name. Possible solutions are to either add a VisuallyHidden with "Select" as the content to the columnheader, or set the role for the columnheader to gridcell, when the Select All Checkbox is hidden.
3. Documentation examples include unlabelled icon Button controls for illustrative purposes. ideally we should demonstrate best practices for labelling these buttons.

Closes #2132

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue 2132](https://github.com/adobe/react-spectrum/issue/2132).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 🧢 Your Project:

Adobe/Accessibility

### CODE REVIEW

**Review by LFDanLu — COMMENTED** [review]
Tested locally, confirmed the localization of the Select/Select All checkboxes and that VoiceOver now announces "Select dimmed" for the hidden select all checkbox for single selection TableViews.

I assume the second approach of "set the role for the columnheader to gridcell" would involve doing `role: isSelectionCellDisabled? 'gridcell' : 'columnheader',` in `useTableColumnHeader`? Would we still need the the visually hidden component so VoiceOver announces "Select dimmed" in that case? Trying to understand the pros/cons of the 2nd approach vs this one

File: packages/@react-aria/table/intl/ar-AE.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "Select": "يحدد",
+  "Select All": "تحديد الكل"
```
[the remark this answers is not in the document]
Comment (majornista) [reply packages/@react-aria/table/intl/ar-AE.json:3]: Like "select" and "selectAll"?

**Comment by majornista** [comment]
> Tested locally, confirmed the localization of the Select/Select All checkboxes and that VoiceOver now announces "Select dimmed" for the hidden select all checkbox for single selection TableViews.
> 
> I assume the second approach of "set the role for the columnheader to gridcell" would involve doing `role: isSelectionCellDisabled? 'gridcell' : 'columnheader',` in `useTableColumnHeader`? Would we still need the the visually hidden component so VoiceOver announces "Select dimmed" in that case? Trying to understand the pros/cons of the 2nd approach vs this one

- That's correct `role: isSelectionCellDisabled? 'gridcell' : 'columnheader',` in `useTableColumnHeader`. 
- The visually hidden Checkbox seems to be there for layout purposes, so that the contents fill the cell. I wonder if we could do something like `hideHeader`
- I don't think the column header should receive `aria-disabled="true"`, because, other than receiving focus when navigating the table, it is not interactive, and it is the reason why VoiceOver announces "Select dimmed" instead of "Select".

File: packages/@react-aria/table/intl/ar-AE.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (LFDanLu) [reply packages/@react-aria/table/intl/ar-AE.json]: Yep

**Review by LFDanLu — APPROVED** [review]
LGTM

File: packages/@react-spectrum/table/stories/Table.stories.tsx [hunk 2]
```diff
@@ -711,12 +711,11 @@ storiesOf('TableView', module)
           <Row>
             <Cell>One</Cell>
             <Cell>
-              <ActionButton isQuiet>
-                <Add />
+              <ActionButton isQuiet aria-label="Add Info">                <Add />
```
Comment (snowystinger) [inline packages/@react-spectrum/table/stories/Table.stories.tsx:714]: can we undo this formatting?

**Review by devongovett — APPROVED** [review]

**Review by snowystinger — APPROVED** [review]
lgtm

File: packages/@react-spectrum/table/stories/Table.stories.tsx (hunk 2 above)
Comment (devongovett) [reply packages/@react-spectrum/table/stories/Table.stories.tsx:714]:
```suggestion
              <ActionButton isQuiet aria-label="Add Info">
                <Add />
```

**Review by snowystinger — APPROVED** [review]
re-approving, haha, thanks @devongovett

**Review by devongovett — APPROVED** [review]

**Pull request merged**